rpcs3/Emu/Cell/Modules/cellGem.cpp

```diff
@@ -765,9 +898,31 @@ s32 cellGemPrepareVideoConvert(vm::cptr<CellGemVideoConvertAttribute> vc_attribu
 	return CELL_OK;
 }
 
-s32 cellGemReadExternalPortDeviceInfo()
+s32 cellGemReadExternalPortDeviceInfo(u32 gem_num, vm::ptr<u32> ext_id, vm::ptr<u8[CELL_GEM_EXTERNAL_PORT_DEVICE_INFO_SIZE]> ext_info)
 {
-	UNIMPLEMENTED_FUNC(cellGem);
+	cellGem.todo("cellGemReset(gem_num=%d, ext_id=*0x%x, ext_info=%s)", gem_num, ext_id, ext_info);
+	auto gem = fxm::get<gem_t>();
+
+	if (!gem)
+	{
+		return CELL_GEM_ERROR_UNINITIALIZED;
+	}
+
+	if (!check_gem_num(gem_num))
```

Comment:
should i check for !ext_id and !ext_info ?

Reply:
libgem only checks for `ext_id` apparently ![image](https://user-images.githubusercontent.com/6632271/37981050-46c724a8-31f5-11e8-9c2b-f3331912eef2.png)